### PR TITLE
Ensure that ignoreUrlParametersMatching works with directoryIndex

### LINKED ISSUE #758: Using ignoreUrlParametersMatching has a bug when testing with `directoryIndex`
**Library Affected**:
workbox-sw

**Browser & Platform**:
all browsers

**Issue or Feature Request Description**:
Issue: using option `ignoreUrlParametersMatching` does not work as expected.

Expected behavior is for the `service-worker` to serve the following urls `//localhost/` & `//localhost/?launcher=true`. But `//localhost/?launcher=true` doesn't get served from cache despite the `launcher` parameter being added to be ignored.

Used options:
```js
new self.WorkboxSW({
  "skipWaiting": true,
  "ignoreUrlParametersMatching": [/launcher/]
});
```

I believe this is a bug with the `capture` function which tries to add the `directoryIndex` and test against the original `url` instead of the `strippedUrl`.

Suspected culprit line is: https://github.com/GoogleChrome/workbox/blob/master/packages/workbox-sw/src/lib/workbox-sw.js#L329

*When reporting bugs, please include relevant JavaScript Console logs and links to public URLs at which the issue could be reproduced.*
N/A

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @addyosmani @gauntface

Fixes #758

This fixes a small bug due to using the wrong variable inside of our WorkboxSW precache route.

### CODE REVIEW

**Comment by jeffposnick** [comment]
Something about the new test is causing Firefox to crash... tracking that down now.

**Comment by jeffposnick** [comment]
I've added some additional comments and refactored some of the variables, and hopefully it's mergeable now. I had initially tried to come up with a more straightforward way of testing the actual functionality, and what I ended up with was about as direct as I could figure out.

(Stubbing out some of the `Route` methods didn't seem possible, as they're not exposed anywhere in the public interface for `WorkboxSW`.)

File: packages/workbox-sw/test/sw/directory-index-custom.js [hunk 1]
```diff
@@ -59,4 +59,45 @@ describe(`Test Directory Index`, function() {
...
+    return new Promise((resolve, reject) => {
+      const fetchEvent = new FetchEvent('fetch', {
+        // Request the URL without the directory index, but with the URL param.
+        request: new Request(urlToRequest),
+      });
+      fetchEvent.respondWith = (promiseChain) => {
+        promiseChain.then(() => {
+          // Make sure that the cache.match() stub was called, indicating that
+          // there was a match against the precached URL.
+          if (stub.called) {
```
Comment (gauntface) [inline packages/workbox-sw/test/sw/directory-index-custom.js]: What is caches.match called with? Should it be with or without the directoryIndex value or not?

File: packages/workbox-sw/test/sw/directory-index-custom.js [hunk 2]
```diff
@@ -59,4 +59,45 @@ describe(`Test Directory Index`, function() {
...
+      ignoreUrlParametersMatching,
+    });
+    workboxSW.precache([urlToPrecache]);
+
+    return new Promise((resolve, reject) => {
+      const fetchEvent = new FetchEvent('fetch', {
+        // Request the URL without the directory index, but with the URL param.
+        request: new Request(urlToRequest),
+      });
+      fetchEvent.respondWith = (promiseChain) => {
```
Comment (gauntface) [inline packages/workbox-sw/test/sw/directory-index-custom.js]: Can you stub this out? Would make it a little easier to read  / follow if you could

File: packages/workbox-sw/test/sw/directory-index-custom.js (hunk 1 above)
Comment (jeffposnick) [reply packages/workbox-sw/test/sw/directory-index-custom.js]: Added in a check.

File: packages/workbox-sw/test/sw/directory-index-custom.js (hunk 2 above)
Comment (jeffposnick) [reply packages/workbox-sw/test/sw/directory-index-custom.js]: I don't think I can stub out `respondWith()` and still get things to work, no. (I copied this pattern from other tests in this suite, e.g. the previous test in this file.)

**Pull request merged**